specification/botservice/resource-manager/Microsoft.BotService/preview/2017-12-01/botservice.json

```diff
@@ -1618,6 +1982,202 @@
         "siteName"
       ]
     },
+    "ConnectionItemName": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Connection Item name that has been added in the API",
+          "readOnly": true
+        }
+      },
+      "description": "The display name of a connection Item Setting registered with the Bot"
+    },
+    "ConnectionSettingParameter": {
+      "properties": {
+        "key": {
+          "type": "string",
+          "description": "Key for the Connection Setting Parameter.",
+          "readOnly": true
+        },
+        "value": {
+          "type": "string",
+          "description": "Value associated with the Connection Setting Parameter.",
+          "readOnly": true
+        }
+      },
+      "description": "Extra Parameter in a Connection Setting Properties to indicate service provider specific properties"
+    },
+    "ConnectionSettingProperties": {
+      "properties": {
+        "clientId": {
+          "type": "string",
+          "description": "Client Id associated with the Connection Setting."
+        },
+        "clientSecret": {

```

Comment:
should not returned in the response of GET, PUT, PATCH. just confirming.

Reply:
Yes. Also, i added a new listwithsecrets path, similar to how we have channel/listwithkeys method. On doing a post to this url, the caller can get a particular connectionname with the secrets present.
Otherwise, the clientsecret will not be returned in a normal get,put, patch